query-coordinator/src/main/scala/com/socrata/querycoordinator/rollups/ExpressionRewriter.scala

```diff
@@ -108,9 +108,9 @@ class ExpressionRewriter(val rollupColumnId: (Int) => String,
   }
 
   /** Find the first column index that is a count(*) or count(literal). */
-  private def findCountStarOrLiteral(rollupColIdx: Map[Expr, Int]): Option[Int] = {
+  private def findCountStarOrLiteral(fc: FunctionCall, rollupColIdx: Map[Expr, Int]): Option[Int] = {
     rollupColIdx.find {
-      case (fc: FunctionCall, _) => isCountStarOrLiteral(fc)
+      case (e: FunctionCall, _) if e.filter == fc.filter => isCountStarOrLiteral(e)
```

Comment:
This seems reasonable.
Presumably the filter can be rewritten so that it matches despite not being a 1:1 equivalence.
i.e  `filter where n > 2 and n > 4` is equivalent to `filter where n == 3`

Reply:
But let's see how far we can get without it.